src/ILCompiler/src/Program.cs

```diff
@@ -305,10 +305,18 @@ private int Run(string[] args)
             if (_isWasmCodegen)
                 _targetArchitecture = TargetArchitecture.Wasm32;
 
+            bool supportsReflection = !_disableReflection && !_isReadyToRunCodeGen && _systemModuleName == DefaultSystemModule;
+
             //
             // Initialize type system context
             //
 
+            // If reflection is not supported, some of the delegate features don't make sense to include.
+            // We can't create open instance delegates, and there's no LINQ expression interpreter.
+            var delegateInfoCreator = supportsReflection ?
+                (DelegateInfoCreator)(delegateType => new DelegateInfo(delegateType, DelegateFeature.All)) :
+                (DelegateInfoCreator)(delegateType => new DelegateInfo(delegateType, DelegateFeature.DynamicInvoke));
```

Comment:
Do we want to have `DynamicInvoke` enabled when reflection is disabled? `DynamicInvoke` sounds like a reflection to me. Does it even work with reflection disabled?

Reply:
`DynamicInvoke` works with reflection disabled. If we remove it, we get extra 16 kB back on Hello World, so that's nice, but not that much.

But we'll have to make Delegate.DynamicInvoke be able to handle the thunk being null.

I can remove that too if you think it's not useful. I'm on the fence.

Reply:
I do not think it is useful. I cannot think about a scenario where somebody would use DynamicInvoke and avoided all other reflection. I would remove it and see whether it is going to show up anywhere.